packages/rocketchat-search/server/events/events.js

```diff
@@ -0,0 +1,59 @@
+import {searchProviderService} from '../service/providerService';
+import SearchLogger from '../logger/logger';
+
+class EventService {
+
+	/*eslint no-unused-vars: [2, { "args": "none" }]*/
+	_pushError(name, value, payload) {
+		//TODO implement a (performant) cache
+		SearchLogger.debug(`Error on event '${ name }' with id '${ value }'`);
+	}
+
+	promoteEvent(name, value, payload) {
+		if (!(searchProviderService.activeProvider && searchProviderService.activeProvider.on(name, value, payload))) {
+			this._pushError(name, value, payload);
+		}
+	}
+}
+
+const eventService = new EventService();
+
+/**
+ * Listen to message changes via Hooks
+ */
+RocketChat.callbacks.add('afterSaveMessage', function(m) {
+	eventService.promoteEvent('message.save', m._id, m);
+});
+
+RocketChat.callbacks.add('afterDeleteMessage', function(m) {
+	eventService.promoteEvent('message.delete', m._id);
+});
+
+/**
+ * Listen to user and room changes via cursor
+ */
+const cursor = Meteor.users.find({}, {fields: {name:1, username:1, emails:1, active:1}});
```

Comment:
I'm not sure about the resources used by long living open cursors. At least for both (users and rooms) cursors I know we have an `EventEmitter` on server's cache layer that could be used here.

Reply:
I will have a look at the Emitter, I also don‘t like the cursors here but wasn‘t aware of that way

Reply:
you can see [ModelsCache](https://github.com/RocketChat/Rocket.Chat/blob/Hudell-saml-users-must-not-have-pass-change-option/packages/rocketchat-lib/server/models/_BaseCache.js#L125) being an `EventEmitter`.. so models on cache have many events you can listen on.